### PR TITLE
feat: verify contract API implementation

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

fixes: #22
fixes: APE-35
Allows user to verify contracts.

Required PR(s):

https://github.com/ApeWorX/ape/pull/1026
https://github.com/ApeWorX/ape/pull/1008

### How I did it

Carefully following their API, cross-checking with browser and Hardhat's implementation and Brownie until things worked.
Had to make changes in `ape-solidity` and `ape` to get it to work.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: ape_etherscan/verify.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            constructor_arguments=self.constructor_arguments,
+            evm_version=evm_version,
+            license_type=self.license_code,
+        )
+        self._wait_for_verification(guid)
+
+    def _wait_for_verification(self, guid: str):
+        iterations = 0
+        timeout = 100
+        while iterations < 25:
```
Comment (fubuloubu) [inline ape_etherscan/verify.py:151]: would probably be nicer as a for loop

Comment (fubuloubu) [reply ape_etherscan/verify.py:151]: Maybe sometthing like:

```py
for iteration in range(num_iterations):
    ...
    time.sleep(2.5)
else:
    raise  # timeout error
```

File: ape_etherscan/verify.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                uri = self.provider.network.explorer.get_address_url(self.address)
+                logger.success(f"Contract verification successful!\n{uri}#code")
+                break
+
+            status_message = f"Contract verification status: {verification_update}"
+            logger.info(status_message)
+
+            time.sleep(2.5)
+            iterations += 1
+            if iterations == timeout:
```
Comment (fubuloubu) [inline ape_etherscan/verify.py:170]: if I'm reading this correctly, this will never be true (iterations cannot exceed 25, while timeout is 100)

**Comment by unparalleled-js** [comment]
see https://github.com/ApeWorX/ape-etherscan/pull/42

**Pull request closed without merging**